tests/internal/test_context_events_api.py

```diff
@@ -64,6 +65,7 @@ def listener():
 
                 core.on(event_name, listener)
 
+            sleep(make_target_id * 0.0001)  # ensure threads finish in order
```

Comment:
It's possible that synchronizing the threads by using a `threading.Event` would accomplish the same goal more reliably.

Reply:
OK, I think I'll do it in a follow-up PR